packages/core/test/ts_typings/dispatcher-typings.test.ts

```diff
@@ -62,6 +62,50 @@ describe('Dispatcher typings', () => {
 			dispatch.myModel.inc(4)
 			dispatch.myModel.inc()
 		})
+
+		it('payload defined as any type', () => {
+			const model = createModel<RootModel>()({
+				state: 0,
+				reducers: {
+					// eslint-disable-next-line @typescript-eslint/no-inferrable-types
+					inc(state, payload: any) {
+						return state + payload
+					},
+				},
+			})
+			interface RootModel extends Models<RootModel> {
+				myModel: typeof model
+			}
+
+			const store = init({ models: { myModel: model } })
+
+			const { dispatch } = store
+			dispatch.myModel.inc(4)
+			dispatch.myModel.inc('4')
+			// @ts-expect-error
+			dispatch.myModel.inc()
```

Comment:
mate @semoal we need to differentiate the `payload?: any` from `payload: any`. Or better, `payload?: number` from `payload: number | undefined`. In our current inference, we treat them the same way, users can use an optional payload. 
There wouldn't be any run-time errors thrown, but it's inconsistent with TypeScript bahevior.